easybuild/easyconfigs/r/RStan/RStan-2.21.5-foss-2021b-R-4.1.2.eb

```
@@ -0,0 +1,55 @@
+easyblock = 'Bundle'
+
+name = 'RStan'
+version = '2.21.5'
+versionsuffix = '-R-%(rver)s'
+
+homepage = 'https://github.com/stan-dev/rstan'
+description = """RStan is the R interface to Stan. Stan is a state-of-the-art
+ platform for statistical modeling and high-performance statistical computation."""
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+
+dependencies = [
+    ('R', '4.1.2')
+]
+
+# preinstallopts = 'export PKG_CXXFLAGS="$CXXFLAGS" && '
+# preinstallopts = 'export PKG_CXX14FLAGS="$CXX14FLAGS -DSTAN_THREADS" &&'
+
+exts_defaultclass = 'RPackage'
+exts_filter = ("R -q --no-save", "library(%(ext_name)s)")
+
+exts_default_options = {
+    'source_urls': [
+        'https://cran.r-project.org/src/contrib',
+        'https://cran.r-project.org/src/contrib/Archive/%(name)s',
+    ],
+    'source_tmpl': '%(name)s_%(version)s.tar.gz',
+}
+
+exts_list = [
+    ('inline', '0.3.19', {
+        'checksums': ['0ee9309bb7dab0b97761ddd18381aa12bd7d54678ccd7bec00784e831f4c99d5'],
+    }),
+    ('loo', '2.5.1', {
+        'checksums': ['866a2f54a4e8726cc3062e27daa8a073e6ac4aeb6719af7845284f7a668745f1'],
+    }),
+    ('StanHeaders', '2.21.0-7', {
+        'installopts': '--configure-args="CXXFLAGS=\"$CXXFLAGS -DSTAN_THREADS\""',
+        'checksums': ['27546e064f0e907e031d9185ad55245d118d82fbe3074ecb1d76fae8b9f2336b'],
+    }),
+    ('rstan', version, {
+        'installopts': '--configure-args="CXXFLAGS=\"$CXXFLAGS -DSTAN_THREADS\""',
```

I'm seeing this in the output:
```
== 2022-06-11 13:32:34,015 run.py:623 INFO cmd " R CMD INSTALL /dev/shm/branfosj/build-up-EL8/RStan/2.21.5/foss-2021b-R-4.1.2/rstan/rstan   --library=/rds/projects/2017/branfosj-rse/easybuild/EL8-cas/software/RStan/2.21.5-foss-2021b-R-4.1.2/ --no-clean-on-error --configure-args="CXXFLAGS="$CXXFLAGS -DSTAN_THREADS""" exited with exit code 0 and output:
Warning: unknown option -ftree-vectorize
Warning: unknown option -march=native
Warning: unknown option -fno-math-errno
Warning: unknown option -DSTAN_THREADS
```

So, I am not sure this is the right method to get this option in.

---

Sorry to be so late getting back to this.  I'm a bit confused about the error.  Surely  ````-march=native```` is a standard ````gcc```` option - I don't see how that can be unknown unless it is getting passed to something other than ````gcc````.

---

This is probably because of the double double quotes in `--configure-args="CXXFLAGS="$CXXFLAGS -DSTAN_THREADS""`. You can try replacing the outer ones with single quotes, or maybe the inner ones can be removed?

```suggestion
        'installopts': '--configure-args=\'CXXFLAGS="$CXXFLAGS -DSTAN_THREADS"\'',
```
or
```suggestion
        'installopts': '--configure-args="CXXFLAGS=$CXXFLAGS -DSTAN_THREADS"',
```